### DISCUSSION TITLE
Typst CLI shows compilation's "time value" and "time unit" without whitespace between them

### DISCUSSION THREAD

**Discussion by Andrew15-5** [typst/typst#2100] [General]
I'm not 100% sure, but putting time units directly after time values isn't grammatically correct: `2s`, `341ms`. If I'm right not this, then Typst should add a whitespace between time value and time unit: `2 s`, `341 ms`.

P.S. I've seen a lot of examples where there is no whitespace between, but as far as grammar goes, I think this is just wrong.

**Update**:
It appears that the problem is with the `std::time::Duration` in Rust itself and not with the Typst. So opening an issue here won't do much.

**Comment by LaurenzV** [comment]
I can't say whether it really is grammatically incorrect strictly speaking, but I definitely have seen this a lot in other contexts as well, and in my opinion it looks better without whitespace. But I suppose it's a matter of personal preference.

**Reply by Andrew15-5** [reply]
I, too, think that without the whitespace it looks better, but I think it's entirely because I've seen many people omit the whitespace. If I haven't seen it, it would be very weird, because I don't think there are any other "popular" whitespace between "words" omitting examples. Maybe `i.e.`/`e.g.`, but this is very different (AFAIK it's Latin and an acronym with dots).

**Comment by Enivex** [comment]
> I can't say whether it really is grammatically incorrect strictly speaking, but I definitely have seen this a lot in other contexts as well, and in my opinion it looks better without whitespace. But I suppose it's a matter of personal preference.

It's not really a matter of personal preference. The standard rules for units say that there has to be a space there

https://www.bipm.org/documents/20126/41483022/SI-Brochure-9.pdf/

5.4.3 on page 149: "The numerical value always precedes the unit and a space is always used to separate the
unit from the number."

**Comment by uben0** [comment]
You are right. Most institutions that publish recommendations about grammar correctness does recommend a space between the number and the unit.

**Comment by Andrew15-5** [comment]
I can also refer to my school, where during Physics class we never even thought of omitting the whitespace. Maybe this issue is only present in digital form, but not in written. It certainly looks like it.
